### ISSUE TITLE
Fix S3949 FN by adding support for constants in intermediate code generation

### ISSUE THREAD

**Issue by loic-joly-sonarsource** [SonarSource/sonar-dotnet#3368]
### Description

When detecting integer overflows, we currently don't support constants, such as `int.MaxValue`, and therefore don't trigger the rule in situations where it should be triggered.

### Repro steps
Related to [community thread](https://community.sonarsource.com/t/rule-s3949-fn-calculations-should-not-overflow-bug-warning-is-not-caught-by-sonar-scanner-and-it-is-not-displayed-in-the-report-on-sonarqube-8-2-version-community-edition/21883)

### Actual behavior

Looking at internal intermediate code representation, we generate: 
```
%5 = cbde.unknown : i32  // int (PredefinedType)
%6 = cbde.unknown : i32  // int.MaxValue (SimpleMemberAccessExpression)
```
While we should generate:
```
%5 = constant 2147483647 : i32
```
